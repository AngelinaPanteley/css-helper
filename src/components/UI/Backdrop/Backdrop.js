import React from 'react';
import PropTypes from 'prop-types';
import styles from './Backdrop.scss';

const backdrop = (props) => {
  const classes = [styles.Backdrop];
  if (props.withBackground) {
    classes.push(styles.With_Background);
  }
  return (
    props.show
      ? <div className={classes.join(' ')} onClick={props.clicked}></div>
      : null
  )
};

backdrop.propTypes = {
  withBackground: PropTypes.bool,
  clicked: PropTypes.func.isRequired,
}

export default backdrop;