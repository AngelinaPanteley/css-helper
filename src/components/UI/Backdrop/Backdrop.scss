@import '../../../variables';

.Backdrop {
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: map-get($z-index, backdrop);
  left: 0;
  top: 0;

  &.With_Background {
    background-color: $color-backdrop-bg;
  }
}