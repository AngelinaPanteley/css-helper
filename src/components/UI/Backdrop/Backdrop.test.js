"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';

import Backdrop from './Backdrop.js';

configure({ adapter: new Adapter() });

describe('>>>Backdrop', () => {

  //SNAPSHOTS

  const props = {
    show: true,
    withBackground: true,
    clicked: () => { },
  };

  const backdropComponent = <Backdrop {...props} />;

  const component = renderer.create(backdropComponent);

  it('renders itself correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });
})