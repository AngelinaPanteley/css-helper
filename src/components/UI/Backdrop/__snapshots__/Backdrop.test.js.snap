// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Backdrop renders itself correctly 1`] = `
<div
  className=" "
  onClick={[Function]}
/>
`;
